src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/CreateIncentive/CreateIncentiveCommandHandler.cs

```diff
@@ -7,28 +7,39 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
-using SFA.DAS.EmployerIncentives.Enums;
+using Microsoft.EntityFrameworkCore.Internal;
+using SFA.DAS.EmployerIncentives.Commands.Types.IncentiveApplications;
 
 namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.CreateIncentive
 {
     public class CreateIncentiveCommandHandler : ICommandHandler<CreateIncentiveCommand>
     {
         private readonly IApprenticeshipIncentiveFactory _apprenticeshipIncentiveFactory;
         private readonly IApprenticeshipIncentiveDomainRepository _apprenticeshipIncentiveDomainRepository;
+        private readonly ICommandPublisher _commandPublisher;
 
         public CreateIncentiveCommandHandler(
             IApprenticeshipIncentiveFactory apprenticeshipIncentiveFactory,
-            IApprenticeshipIncentiveDomainRepository apprenticeshipIncentiveDomainRepository)
+            IApprenticeshipIncentiveDomainRepository apprenticeshipIncentiveDomainRepository,
+            ICommandPublisher commandPublisher)
         {
             _apprenticeshipIncentiveFactory = apprenticeshipIncentiveFactory;
             _apprenticeshipIncentiveDomainRepository = apprenticeshipIncentiveDomainRepository;
+            _commandPublisher = commandPublisher;
```

Comment:
You might be able to refactor this so the save repo method is still called as that'll publish the messages (I think!).  That way you can avoid the dependency.

Edit: this might be a terrible suggestion actually as you'd need to publish the EarningsCalculated domain event instead.